src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/commands/mobile/subscribe_vehicle_data_request.cc

```diff
@@ -0,0 +1,506 @@
+    if (hmi_request.func_id == event.id()) {
+      hmi_request.status = static_cast<hmi_apis::Common_Result::eType>(
+          message[strings::params][hmi_response::code].asInt());
+      if (hmi_apis::Common_Result::SUCCESS == hmi_request.status)
+        hmi_request.value = message[strings::msg_params][hmi_request.str];
+      hmi_request.complete = true;
+      break;
+    }
+  }
+  bool all_complete = true;
+  bool any_arg_success = false;
+  mobile_api::Result::eType status = mobile_api::Result::eType::SUCCESS;
+  for (HmiRequests::const_iterator it = hmi_requests_.begin();
+       it != hmi_requests_.end();
+       ++it) {
+    if (!it->complete) {
+      all_complete = false;
+      break;
+    }
+    if (hmi_apis::Common_Result::SUCCESS != it->status) {
+      if (mobile_api::Result::SUCCESS == status) {
+        status = static_cast<mobile_apis::Result::eType>(it->status);
+      } else if (status !=
+                 static_cast<mobile_apis::Result::eType>(it->status)) {
+        status = mobile_api::Result::eType::GENERIC_ERROR;
+      }
+      LOG4CXX_TRACE(logger_,
+                    "Status from HMI: " << it->status
+                                        << ", so response status become "
+                                        << status);
+    } else {
+      any_arg_success = true;
+    }
+  }
+
+  if (all_complete) {
+    smart_objects::SmartObject response_params(smart_objects::SmartType_Map);
+    if (any_arg_success) {
+      for (HmiRequests::const_iterator it = hmi_requests_.begin();
+           it != hmi_requests_.end();
+           ++it) {
+        response_params[it->str] = it->value;
+      }
+    }
+    LOG4CXX_DEBUG(logger_, "All HMI requests are complete");
+    const bool result = any_arg_success;
+    SendResponse(any_arg_success, status, NULL, &response_params);
+    if (result) {
+      app->UpdateHash();
+    }
+  }
+#else
+
+  hmi_apis::Common_Result::eType hmi_result =
+      static_cast<hmi_apis::Common_Result::eType>(
+          message[strings::params][hmi_response::code].asInt());
+  std::string response_info;
+  GetInfo(message, response_info);
+  const bool result = PrepareResultForMobileResponse(
+      hmi_result, HmiInterfaces::HMI_INTERFACE_VehicleInfo);
+
+  bool is_succeeded = result || !vi_already_subscribed_by_another_apps_.empty();
+
+  mobile_apis::Result::eType result_code =
+      MessageHelper::HMIToMobileResult(hmi_result);
+
+  if (is_succeeded) {
+    if (!vi_already_subscribed_by_this_app_.empty()) {
+      result_code = mobile_apis::Result::IGNORED;
+      response_info = "Already subscribed on some provided VehicleData.";
+    }
+
+    if (!vi_waiting_for_subscribe_.empty()) {
+      LOG4CXX_DEBUG(logger_, "Subscribing to all pending VehicleData");
+      app_mngr::VehicleInfoSubscriptions::const_iterator key =
+          vi_waiting_for_subscribe_.begin();
+      for (; key != vi_waiting_for_subscribe_.end(); ++key) {
+        app->SubscribeToIVI(*key);
```

Comment:
SubscribeToIVI shall be a function in new plugin

Reply:
@yang1070 fixed 0afeef687727606c83f2627971a614f671005bf4